web/packages/teleterm/src/services/tshd/types.ts

```diff
@@ -46,7 +46,7 @@ export interface Server extends apiServer.Server.AsObject {
 
 export interface Gateway extends apiGateway.Gateway.AsObject {
   uri: uri.GatewayUri;
-  targetUri: uri.DatabaseUri;
+  targetUri: uri.DatabaseUri | uri.KubeUri;
```

Comment:
I rewrote the logic a bit since short is same as long, and target name requires different parsing for kube. not sure if there is an easier way.
```
      let targetName: string;
      let targetUser: string;
      let targetDesc: string;

      // Try to get target name and user from gateway object.
      if (gateway) {
        targetName = gateway.targetName;
        targetUser = gateway.targetUser;
      }

      // Try to get target name from target URI.
      if (!targetName) {
        targetName = routing.parseDbUri(targetUri)?.params['dbId'];
      }
      if (!targetName) {
        targetName = routing.parseKubeUri(targetUri)?.params['kubeId'];
      }

      if (targetUser) {
        targetDesc = `${targetName} as ${targetUser}`;
      } else {
        targetDesc = targetName;
      }
```

Reply:
Nice, I allowed myself to modify it a little bit so that we use `targetUri` as a fallback if all else fails.

During normal operation this shouldn't happen, but if there's a bug in tshd and we pass a wrong URI or something, it'll result in a better error message.